pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,776 @@
+/*
+Copyright 2022 The Karmada operator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:path=karmadadeployments,shortName=kmd;kmds,scope=Namespaced
```

Comment:
the short name `kmd` might be ambiguous with `k`ar`m`a`d`a.

Reply:
If the CRD kind name is short, I think we  should not need the short-name.

Reply:
I agree, I think we can remove the shortName for now, since we are still considering the Kind name.

Reply:
@RainbowMango done.  

Naming is harder than coding. 😄

Reply:
@RainbowMango any suggests?